packages/waku/src/router/create-pages.ts

```diff
@@ -457,3 +458,277 @@ export function createPages<
+    } else {
+      throw new Error('Invalid page configuration');
+    }
+    return page as Exclude<typeof page, { path: never } | { render: never }>;
+  };
+
+  const createLayout: CreateLayout = (layout) => {
+    if (configured) {
+      throw new Error('createLayout no longer available');
+    }
+    if (layout.render === 'static') {
+      const id = joinPath(layout.path, 'layout').replace(/^\//, '');
+      registerStaticComponent(id, layout.component);
+    } else if (layout.render === 'dynamic') {
+      if (dynamicLayoutPathMap.has(layout.path)) {
+        throw new Error(`Duplicated dynamic path: ${layout.path}`);
+      }
+      const pathSpec = parsePathWithSlug(layout.path);
+      dynamicLayoutPathMap.set(layout.path, [pathSpec, layout.component]);
+    } else {
+      throw new Error('Invalid layout configuration');
+    }
+  };
+
+  const createRoot: CreateRoot = (root) => {
+    if (configured) {
+      throw new Error('createRoot no longer available');
+    }
+    if (rootItem) {
+      throw new Error(`Duplicated root component`);
+    }
+    if (root.render === 'static' || root.render === 'dynamic') {
+      rootItem = root;
+    } else {
+      throw new Error('Invalid root configuration');
+    }
+  };
+
+  let ready: Promise<AllPages | void> | undefined;
+  const configure = async () => {
+    if (!configured && !ready) {
+      ready = fn({ createPage, createLayout, createRoot });
+      await ready;
+      configured = true;
+    }
+    await ready;
+  };
+
+  const definedRouter = new_defineRouter({
+    getPathConfig: async () => {
+      await configure();
+      const paths: {
+        pattern: string;
+        path: PathSpec;
+        components: Record<string, { isStatic: boolean }>;
+        noSsr: boolean;
+      }[] = [];
+      for (const [path, pathSpec] of fixedPathSet) {
+        const noSsr = noSsrSet.has(pathSpec);
+        const isStatic = (() => {
+          for (const [_, [layoutPathSpec]] of dynamicLayoutPathMap) {
+            if (hasPathSpecPrefix(layoutPathSpec, pathSpec)) {
+              return false;
+            }
+          }
+          return true;
+        })();
+
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: { 'layout:/': { isStatic } },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of dynamicPagePathMap) {
+        const noSsr = noSsrSet.has(pathSpec);
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: { 'page:/': { isStatic: false } },
+          noSsr,
+        });
+      }
+      for (const [path, [pathSpec]] of wildcardPagePathMap) {
+        const noSsr = noSsrSet.has(pathSpec);
+        paths.push({
+          pattern: path2regexp(parsePathWithSlug(path)),
+          path: pathSpec,
+          components: { 'page:/': { isStatic: false } },
```

Comment:
This doesn't seem correct. It should include all components for a "page". It's like all components from the current `getComponentIds`.